### ISSUE TITLE
Add macros (#define) for rename function

### ISSUE THREAD

**Issue by KennyDalglish** [pBlueG/SA-MP-MySQL#114]
Hi. You rename function in mysql r40. Please, add macros (#define) for old name function.

**Comment by Jeroen52** [comment]
If your script is not compatible with R40, make it compatible or use R39-5.

**Comment by maddinat0r** [comment]
Just like @Jeroen52 said, make it compatible by rewriting your code or adding those macros yourself in your code, or stay on R39-6.

**Comment by KennyDalglish** [comment]
@maddinat0r Not a good idea to edit someone else's code (and even more updated). I don't understand why you cannot make a #define? What was the renamed functions?

**Comment by Jeroen52** [comment]
@KennyDalglish R40 breaks compatibility with older code, if you'd like to run this plugin with older code you need to either rewrite some of your code, or use R39-6.
For a full list of changes, visit https://github.com/pBlueG/SA-MP-MySQL/releases/R40 for the changelog, as you can see on that page the new version is code-breaking.

**Comment by KennyDalglish** [comment]
So why lose compatibility with the old version? What profit? The list of changes I saw

**Comment by Jeroen52** [comment]
@KennyDalglish A person that I know who upgraded from R39-* to R40 has seen a speed boost when doing certain queries.
````
[1402 miliseconds] | Supposedly R39-*
[385 miliseconds] | Supposedly R40
````
What I have generally heard about R40 is that it is written in a different way, requires different dependencies and is in certain ways more efficient than previous revisions.

You could always keep using R39-6+ until support for it stops or when you upgrade, that is what I'm going to do because my server runs on an old-stable version of CentOS, which is CentOS 6.
If my distro allowed me to upgrade to R40 and above, I'd do it in a heartbeat.

**Comment by IstuntmanI** [comment]
@maddinat0r Can we have some speed comparisons between R36-6 and R40 ?

I guess that the reason maddinat0r has for not adding conversation macros is pretty simple: it isn't hard to update (well, only on huge scripts, like I have). If you don't want to use the newer features just stay on older versions. He wanted to make all functions with the same returning logic (error code) and with similar names, depending on the type of column retrieving you want: index or name. Macros would reduce these benefits. But that's just my guess.

**Comment by Dayvison** [comment]
@KennyDalglish you just need regex, in the topic explain how to update you script many regex have added.

@IstuntmanI the r41 under development, if you want, build yourself.
But if you want win32 binaries take [artifacts](https://ci.appveyor.com/project/maddinat0r/sa-mp-mysql/branch/master/job/v6lfoj8a4f9nqtlp/artifacts) from appveyor.

**Comment by IstuntmanI** [comment]
No, I don't want it. I didn't even updated to R40 yet. I just asked because it seems logical that when a revision number is pushed the entire revision is ready for publishing, but as we can see it isn't having too many commits and it isn't put to releases.

**Comment by maddinat0r** [comment]
> Can we have some speed comparisons between R36-6 and R40 ? Also, I see that on repository you changed the current version name to R41, is it ready ? I see that you didn't add any download.

R40 should be generally a little bit slower, because of the extensive debug mechanics I added (debug logging, debug information from the script to show the line number and file in log calls).
And regarding the revision number: I used to plan increasing version/revision numbers just before releasing a new update. However, I way too often forgot to actually do that. So I just push the number directly whenever I remember.
